lib/ruby_lsp/tapioca/addon.rb

```diff
@@ -23,15 +23,15 @@ class Addon < ::RubyLsp::Addon
       def initialize
         super
 
-        @global_state = T.let(nil, T.nilable(RubyLsp::GlobalState))
         @rails_runner_client = T.let(nil, T.nilable(RubyLsp::Rails::RunnerClient))
+        @index = T.let(nil, T.nilable(RubyIndexer::Index))
```

Comment:
I suggest we continue storing the global state. It contains every important information from the LSP, like configurations, encoding and so on. While we currently only use the index, we may need to access other aspects of it.